applications/luci-app-tailscale/htdocs/luci-static/resources/view/tailscale/setting.js

```diff
@@ -0,0 +1,265 @@
+                })
+        )];
+    });
+}
+
+function getStatus() {
+	var status = {
+		isRunning: false,
+		backendState: undefined,
+		authURL: undefined,
+		displayName: undefined,
+		onlineExitNodes: [],
+		subnetRoutes: []
+	};
+	return Promise.resolve(callServiceList('tailscale')).then(res => {
+		try {
+			status.isRunning = res['tailscale']['instances']['instance1']['running'];
+		} catch (e) {
+			return status;
+		}
+		return fs.exec("/usr/sbin/tailscale", ["status", "--json"]).then(res => {
+			const tailscaleStatus = JSON.parse(res.stdout.replace(/("\w+"):\s*(\d+)/g, '$1:"$2"'));
+			if (!tailscaleStatus.AuthURL && tailscaleStatus.BackendState === "NeedsLogin") {
+				fs.exec("/usr/sbin/tailscale", ["login"]);
+			}
+			status.backendState = tailscaleStatus.BackendState;
+			status.authURL = tailscaleStatus.AuthURL;
+			status.displayName = (status.backendState === "Running") ? 	tailscaleStatus.User[tailscaleStatus.Self.UserID].DisplayName : undefined;
+			status.onlineExitNodes = Object.values(tailscaleStatus.Peer)
+				.flatMap(peer => (peer.ExitNodeOption && peer.Online) ? [peer.HostName] : []);
+			status.subnetRoutes = Object.values(tailscaleStatus.Peer)
+				.flatMap(peer => peer.PrimaryRoutes || []);
+			return status;
+		});
+	}).catch(() => status);
+}
+
+function renderStatus(isRunning) {
+	var spanTemp = '<em><span style="color:%s"><strong>%s %s</strong></span></em>';
+	var renderHTML;
+	if (isRunning) {
+		renderHTML = String.format(spanTemp, 'green', _('Tailscale'), _('RUNNING'));
+	} else {
+		renderHTML = String.format(spanTemp, 'red', _('Tailscale'), _('NOT RUNNING'));
+	}
+
+	return renderHTML;
+}
+
+function renderLogin(loginStatus, authURL, displayName) {
+	var spanTemp = '<span style="color:%s">%s</span>';
+	var renderHTML;
+	if (loginStatus == "NeedsLogin") {
+		renderHTML = String.format('<a href="%s" target="_blank">%s</a>', authURL, _('Need to log in'));
+	} else if (loginStatus == "Running") {
+		renderHTML = String.format('<a href="%s" target="_blank">%s</a>', 'https://login.tailscale.com/admin/machines', displayName);
+		renderHTML += String.format('<br><a style="color:green" id="logout_button">%s</a>', _('Log out and Unbind'));
+	} else {
+		renderHTML = String.format(spanTemp, 'orange', _('NOT RUNNING'));
+	}
+
+	return renderHTML;
+}
+
+return view.extend({
+	load: function() {
+		return Promise.all([
+			uci.load('tailscale'),
+			getStatus(),
+			getInterfaceSubnets()
+		]);
+	},
+
+	render: function(data) {
+		var m, s, o;
+		var statusData = data[1];
+		var interfaceSubnets = data[2];
+		var onlineExitNodes = statusData.onlineExitNodes;
+		var subnetRoutes = statusData.subnetRoutes;
+
+		m = new form.Map('tailscale', _('Tailscale'), _('Tailscale is a cross-platform and easy to use virtual LAN.'));
+
+		s = m.section(form.TypedSection);
+		s.anonymous = true;
+		s.render = function () {
+			poll.add(function() {
+				return Promise.resolve(getStatus()).then(function(res) {
```

Comment:
Since `getStatus` returns a promise, there's no need wrapping it in one, and:
```suggestion
				const res = await getStatus();
```